src/openforms/prefill/contrib/family_members/filters.py

```diff
@@ -0,0 +1,133 @@
+from datetime import datetime
+from typing import Literal
+
+from openforms.contrib.haal_centraal.clients import get_brp_client
+from openforms.contrib.haal_centraal.clients.brp import NaturalPersonDetails
+from openforms.contrib.haal_centraal.constants import BRPVersions
+from openforms.utils.date import get_today
+
+from .constants import HC_DECEASED_ATTRIBUTES
+
+
+def exclude_deceased_for_haal_centraal(
+    results: list[NaturalPersonDetails],
+    hc_version: Literal[BRPVersions.v13, BRPVersions.v20],
+) -> list[NaturalPersonDetails]:
+    bsn_numbers = [member["bsn"] for member in results]
+
+    with get_brp_client() as client:
+        # v2 of HaalCentraal supports request for multiple bsn numbers
+        if hc_version == BRPVersions.v20:
+            members = client.find_person(
+                bsn_numbers, attributes=HC_DECEASED_ATTRIBUTES[hc_version]
+            )
+        # v1 of HaalCentraal requires separate request for each bsn
+        else:
+            members = []
+            for bsn in bsn_numbers:
+                member = client.find_person(
+                    bsn, attributes=HC_DECEASED_ATTRIBUTES[hc_version]
+                )
+                if member:
+                    members.append(member)
+
+        if not members:
+            return results
+
+        deceased_bsns = [
+            member["burgerservicenummer"]
+            for member in members
+            if isinstance(member, dict) and "overlijden" in member
+        ]
+        eligible_members = [
+            result for result in results if result["bsn"] not in deceased_bsns
+        ]
+        return eligible_members
+
+
+def is_eligible_based_on_age(
```

Comment:
Probably a (calculated) (cached_)property on a natural person pydantic model/dataclass would be easier so you can just check on `person.age`:

```py
age_filtered = [
    person for person in persons
    if min_age <= person.age <= max_age
]
```
and then you can also set `min_age` to `min_age or 0` (in case it's None, just go with the lowest possible value) and do the same for `max_age` with `500` or so - I think it'll be a while before someone actually lives for 5 centuries :)